src/Tickets/Commerce/Gateways/PayPal/WhoDat.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace TEC\Tickets\Commerce\Gateways\PayPal;
+
+/**
+ * Class Connect_Client
+ *
+ * @since   5.1.6
+ *
+ * @package TEC\Tickets\Commerce\Gateways\PayPal
+ */
+class WhoDat {
+
+	/**
+	 * The API URL.
+	 *
+	 * @since 5.1.6
+	 *
+	 * @var string
+	 */
+	protected $api_url = 'https://whodat.theeventscalendar.com/tickets/paypal/connect';
+
+	/**
+	 * Get REST API endpoint URL for requests.
+	 *
+	 * @since TBD
+	 *
+	 *
+	 * @param string $endpoint   The endpoint path.
+	 * @param array  $query_args Query args appended to the URL.
+	 *
+	 * @return string The API URL.
+	 */
+	public function get_api_url( $endpoint, array $query_args = [] ) {
+		return add_query_arg( $query_args, "{$this->api_url}/{$endpoint}" );
+	}
+
+	/**
+	 * Send a POST request to WhoDat inside of the PayPal connection path.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $endpoint
+	 * @param array  $query_args
+	 * @param array  $request_arguments
+	 *
+	 * @return array|null
+	 */
+	public function post( $endpoint, array $query_args = [], array $request_arguments = [] ) {
+		$url = $this->get_api_url( $endpoint, $query_args );
+
+		$default_arguments = [
+			'body'      => [],
+
+			// @todo Remove this when SSL is fixed.
+			'sslverify' => false,
+		];
+		$request_arguments = array_merge_recursive( $default_arguments, $request_arguments );
+		$response          = wp_remote_post( $url, $request_arguments );
+
+		if ( is_wp_error( $response ) ) {
+			tribe( 'logger' )->log_error( sprintf(
+				'[%s] WhoDat request error: %s',
+				$url,
+				$response->get_error_message()
+			), 'whodat-connection' );
+
+			return null;
+		}
+
+		$response = wp_remote_retrieve_body( $response );
+		$response = @json_decode( $response, true );
+
+		if ( ! is_array( $response ) ) {
+			tribe( 'logger' )->log_error( sprintf( '[%s] Unexpected WhoDat response', $url ), 'whodat-connection' );
+
+			return null;
+		}
+
+		return $response;
+	}
+
+	/**
+	 * Retrieves a Partner Link for on-boarding
+	 *
+	 * @param $return_url
+	 * @param $country
+	 *
+	 * @return array|null
+	 */
+	public function get_seller_partner_link( $return_url, $country ) {
```

Comment:
Would recommend ditching this method as it was already implemented via the Onboard class #2039. As we discussed during our call earlier today, this class `WhoDat` will be specific for abstracting the `GET` and `POST` requests.